### ISSUE TITLE
Training image model, the val acc alwasys shows 1

### ISSUE THREAD

**Issue by AlinaXie** [bharathichezhiyan/Shardul_Master_Project#16]
Hi,
Because I read your paper before, I think it is an interesting topic.
So I try to reproduce it.  But when I train the image model, its result is very strange. The validation accuracy always shows 1.
I do not whether there is a problem when I process the code. I have tried many methods but I cannot solve it.
I would be appreciated if I can receive your help. Thanks.
<img width="297" alt="result1" src="https://user-images.githubusercontent.com/46320082/113100726-5bba0280-922e-11eb-8290-f2bbd9d255d4.PNG">
<img width="351" alt="result2" src="https://user-images.githubusercontent.com/46320082/113100743-62e11080-922e-11eb-9785-0359733d2205.PNG">

**Comment by harshgeek4coder** [comment]
Same issue here

**Comment by harshgeek4coder** [comment]
> Hi, Because I read your paper before, I think it is an interesting topic. So I try to reproduce it. But when I train the image model, its result is very strange. The validation accuracy always shows 1. I do not whether there is a problem when I process the code. I have tried many methods but I cannot solve it. I would be appreciated if I can receive your help. Thanks. <img alt="result1" width="297" src="https://user-images.githubusercontent.com/46320082/113100726-5bba0280-922e-11eb-8290-f2bbd9d255d4.PNG"> <img alt="result2" width="351" src="https://user-images.githubusercontent.com/46320082/113100743-62e11080-922e-11eb-9785-0359733d2205.PNG">

@AlinaXie and @bharathichezhiyan  
I somehow managed to resolve this error by refactoring the ```image_model``` function and removing ```GlobalAveragePooling2D()```, and also, modified the image neural network to some extent.
I am mentioning the code for your further reference :

```
def Image_Model(base_model):
  for layer in base_model.layers:
    layer.trainable = False
  x = base_model.output
  return (x)

```

Neural Network : 

```
#MAIN IMAGE MODEL - II : 
def get_image_only_model():
  pre_trained_image_model_vgg16 = VGG16(include_top=False, weights='/content/vgg16_weights_tf_dim_ordering_tf_kernels_notop.h5',input_shape=(224,224) + (3,))
  base_model_loaded = Image_Model(pre_trained_image_model_vgg16)
  flatten_layer = Flatten()(base_model_loaded)
  dropout_layer = Dropout(0.5)(flatten_layer)
  image_pred_layer = Dense(1, activation='sigmoid')(dropout_layer)
  image_only_model = Model(inputs = [pre_trained_image_model_vgg16.input], outputs = image_pred_layer)

  return image_only_model

```

This worked with showing somewhat better learning for VGG16 Model : 

![image](https://user-images.githubusercontent.com/56259814/143229074-31119f8b-53af-4212-8389-ed4ae346ab3c.png)

**Comment by keyuchen21** [comment]
@harshgeek4coder I followed your instruction but still get val_accuracy of 1, could you share your code?

**Comment by Adrillanos** [comment]
Hi, i have a similar issue, the val_acc always show 0.000, any advice?
